include/core/CCompressUtils.h

```diff
@@ -37,50 +37,135 @@ namespace core {
 //!
 class CORE_EXPORT CCompressUtils : private CNonCopyable {
 public:
-    //! The output type
     using TByteVec = std::vector<Bytef>;
+    enum EOperation { E_Deflate, E_Inflate };
```

Comment:
devils advocate: What is the reason to have operations/modes over 2 classes/methods ?

Reply:
It seemed easier to share code this way than have two classes; although I could probably extract common functionality out into a private base class. It also felt at the outset like it was going to be the smaller change to CCompressUtils; although in practice it ended up changing quite a lot. I'll have a think about this alternative.

Reply:
I think this turns out nicer: I added separate implementations with shared base class for inflation and deflation. These can really just wrap up only the small number of library calls which are task specific, so nearly all the implementation is shared.